src/opnsense/mvc/app/controllers/OPNsense/DHCPv6/Api/LeasesController.php

```diff
@@ -0,0 +1,245 @@
+            ];
+            $statics[] = $static;
+        }
+
+        $leases = array_merge($leases, $statics);
+
+        $mac_man = json_decode($backend->configdRun('interface list macdb json'), true);
+        $interfaces = [];
+        foreach ($leases as $idx => $lease) {
+            $leases[$idx]['man'] = '';
+            $leases[$idx]['mac'] = '';
+            /* We infer the MAC from NDP data if available, otherwise we extract it out
+             * of the DUID. However, RFC8415 section 11 states that an attempt to parse
+             * a DUID to obtain a client's link-layer addresss is unreliable, as there is no
+             * guarantee that the client is still using the same link-layer address as when
+             * it generated its DUID. Therefore, if we can link it to a manufacturer, chances
+             * are fairly high that this is a valid MAC address, otherwise we omit the MAC
+             * address.
+             */
+            if (!empty(['address'])) {
+                foreach ($ndp_data as $ndp) {
+                    if ($ndp['ip'] == $lease['address']) {
+                        $leases[$idx]['mac'] = $ndp['mac'];
+                        $leases[$idx]['man'] = empty($ndp['manufacturer']) ? '' : $ndp['manufacturer'];
+                        break;
+                    }
+                }
+            }
+
+            if (!empty($lease['duid'])) {
+                $mac = '';
+                $duid_type = substr($lease['duid'], 0, 5);
+                if ($duid_type === "00:01" || $duid_type === "00:03") {
+                    /* DUID generated based on LL addr with or without timestamp */
+                    $hw_type = substr($lease['duid'], 6, 5);
+                    if ($hw_type == "00:01") { /* HW type ethernet */
+                        $mac = substr($lease['duid'], -17, 17);
+                    }
+                }
+
+                if (!empty($mac)) {
+                    $mac_hi = strtoupper(substr(str_replace(':', '', $mac), 0, 6));
+                    if (array_key_exists($mac_hi, $mac_man)) {
+                        $leases[$idx]['mac'] = $mac;
+                        $leases[$idx]['man'] = $mac_man[$mac_hi];
+                    }
+                }
+            }
+
+            /* include interface */
+            $lease['if_descr'] = '';
+            if (!empty($lease['if'])) {
+                $if = $config->interfaces->{$lease['if']};
+                if (!empty((string)$if->ipaddrv6) && Util::isIpAddress((string)$if->ipaddrv6)) {
+                    $intf = (string)$if->descr;
+                    $leases[$idx]['if_descr'] = $intf;
+                    $leases[$idx]['if'] = $lease['if'];
+
+                    if (!array_key_exists($lease['if'], $interfaces)) {
+                        $interfaces[$lease['if']] = $intf;
+                    }
+                }
+            } else {
+                foreach ($config->dhcpdv6->children() as $dhcpif => $dhcpifconf) {
```

Comment:
As the number of `dhcpdv6` servers is highly likely limited, this shouldn't be an issue, but otherwise you would like to store the interface ranges in a separate loop to prevent a cartesian product.